File: service/api/src/main/java/org/delivery/api/domain/userorder/dto/resp/UserOrderResponse.java
package org.delivery.api.domain.userorder.dto.resp;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import org.delivery.db.order.vo.UserOrderStatus;

public record UserOrderResponse(
        Long id,
        UserOrderStatus status,
        BigDecimal amount,
        LocalDateTime orderedAt,
        LocalDateTime acceptedAt,
        LocalDateTime cookingStartedAt,
        LocalDateTime deliveryStartedAt,
        LocalDateTime receivedAt
) {
}

File: delivery/api/src/main/kotlin/org/delivery/api/domain/userorder/dto/resp/UserOrderResponse.kt
package org.delivery.api.domain.userorder.dto.resp

import org.delivery.db.order.vo.UserOrderStatus
import java.math.BigDecimal
import java.time.LocalDateTime

data class UserOrderResponse(
    var id: Long? = null,
    var status: UserOrderStatus? = null,
    var amount: BigDecimal? = null,
    var orderedAt: LocalDateTime? = null,
    var acceptedAt: LocalDateTime? = null,
    var cookingStartedAt: LocalDateTime? = null,
    var deliveryStartedAt: LocalDateTime? = null,
    var receivedAt: LocalDateTime? = null
)